Issue #1289: Add Application Error event class.

rmouritzen-splunk opened the pull request:
#### Related Issue: 
https://github.com/ocsf/ocsf-schema/issues/1289

#### Description of changes:
Add Application Error event class. See related issue for details.

pagbabian-splunk commented on events/application/application_error.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "uid": 8,
+  "caption": "Application Error",
+  "description": "Application Errors events describe issues with an application that generates OCSF events.",
```
nit: "Application Error events" rather than "Errors". Could also be more general than requiring the application to be on that generates OCSF events.

rmouritzen-splunk replied:
Thanks. Fixed.

floydtree replied:
Agree, we should make the class more generic and not limit it to OCSF translation Apps.

rmouritzen-splunk replied:
IMHO this because too broad and could be confusing. I didn't intent this as way to generically capture a random application error log. If that was the case, it would be call "Application Issue", would clearly indicate that log levels map to severity, and would include a place to capture a stack trace and perhaps other debug information.

I specifically intended this as a way to capture error attempting to map a raw event to OCSF. It _could_ also be used for applications that natively generate OCSF Events, though I don't know what that would look like. I suspect normal application logging would be used in this case.

This class is specifically for the where a raw log cannot be mapped to the OCSF schema, and normal logging isn't approriate as it would require logging the raw event, which in modern cloud environments at least, would leak customer data in to logs (something we try to avoid in the companies I've worked at over the last 10 or so years).

I don't feel the need to be dogmatic here, but if we do want this to be generic, we should have actual use-cases in mind, and so far I'm not seeing it. And because I can't see it, I can't tell if a more generic event would be correct.

rmouritzen-splunk replied:
I updated the description to be more general, but still retain OCSF event generation/handling as an intended case.

pagbabian-splunk commented on events/application/application_error.json:
```diff
@@ -0,0 +1,25 @@
+      "enum": {
+        "1": {
+          "caption": "General Error",
+          "description": "The application generating OCSF events has experienced an error."
```
Maybe just "The application has experienced an error. The details can be found in the <code>message</code> attribute."

rmouritzen-splunk replied:
Fixed. I also changed the other enum value's description for consistency.

pagbabian-splunk commented:
I'm thinking this class can be more generally applicable to application errors, not just those that generate OCSF events - although generating this event would indeed constitute that, but it could also be an application that detects events in some other application that doesn't generate OCSF events. Given you have a `General` `activity_id` we could relax the descriptions a bit.

Also, there should be some direction as to where to put the error message - this is usually the `message` attribute of the Base event class.

rmouritzen-splunk commented:
> I'm thinking this class can be more generally applicable to application errors, not just those that generate OCSF events - although generating this event would indeed constitute that, but it could also be an application that detects events in some other application that doesn't generate OCSF events. Given you have a `General` `activity_id` we could relax the descriptions a bit.
> 
> Also, there should be some direction as to where to put the error message - this is usually the `message` attribute of the Base event class.

@pagbabian-splunk :
I changed the class description to mention placing the error message in the `message` attribute, and I changed the `message` attribute's description to suit this class better.
